framework/src/mesh/MooseMesh.C

```diff
@@ -1193,77 +1194,96 @@ MooseMesh::getBoundaryName(BoundaryID boundary_id)
     return boundary_info.get_nodeset_name(boundary_id);
 }
 
+// specialization for PointListAdaptor<MooseMesh::PeriodicNodeInfo>
+template <>
+inline const Point &
+PointListAdaptor<MooseMesh::PeriodicNodeInfo>::getPoint(
+    const MooseMesh::PeriodicNodeInfo & item) const
+{
+  return *(item.first);
+}
+
 void
 MooseMesh::buildPeriodicNodeMap(std::multimap<dof_id_type, dof_id_type> & periodic_node_map,
                                 unsigned int var_number,
                                 PeriodicBoundaries * pbs) const
 {
-  mooseAssert(!Threads::in_threads,
-              "This function should only be called outside of a threaded "
-              "region due to the use of PointLocator");
-
   TIME_SECTION(_build_periodic_node_map_timer);
 
+  // clear existing map
   periodic_node_map.clear();
 
-  std::unique_ptr<PointLocatorBase> point_locator = getMesh().sub_point_locator();
+  // get periodic nodes
+  std::vector<PeriodicNodeInfo> periodic_nodes;
+  for (const auto & t : getMesh().get_boundary_info().build_node_list())
+  {
+    // unfortunately libMesh does not give us a pointer, so we have to look it up ourselves
+    auto node = _mesh->node_ptr(std::get<0>(t));
```

Comment:
This can be `nullptr` with DistributedMesh in parallel. This might be one of the screwy things that I see with DistributedMesh + PeriodicBCs in some of my runs. You probably don't want to stick NULLs into this data structure.

Reply:
Wait, what? No! `build_node_list` iterates over pointers and returns `ptr->id()`. I have to immediately go back and retrieve `ptr` here. No way this comes back as `nullptr`

Reply:
@jwpeterson says this "should" work but seemed unsure. At the very least you should assert to catch it since it is an assumption.

Reply:
Ok, I added the assertion. Just to be clear, with the _current_ implementation of `build_node_list` this is _guaranteed_ to work. 

https://github.com/libMesh/libmesh/blob/master/src/mesh/boundary_info.C#L1837-L1847

Libmesh _has_ that pointer and returns the result of an `->id()` call on that pointer. My code immediately does the reverse lookup. It could be useful to have a version of that function in libMesh that straight-up returns `Node *` in that vector (by the way, it is unclear to me why a two element tuple is used rather than a pair...).